apps/framework-cli/src/framework/core/infrastructure_map.rs

```diff
@@ -2192,21 +2230,21 @@ impl PartialInfrastructureMap {
                     .expect(not_found);
 
                 let function_process = FunctionProcess {
-                    name: process_id.clone(),
+                    name: process_name.clone(),
                     source_topic_id: source_topic.id(),
                     target_topic_id: Some(target_topic.id()),
                     executable: main_file.to_path_buf(),
                     language,
                     parallel_process_count: target_topic.partition_count,
-                    // TODO pass through version from the TS / PY api
+                    // TODO: what is this version? topic? which one?
                     version: "0.0".to_string(),
```

Comment:
What should this be? Are functions actually versioned?

Reply:
Shouldn't they be versioned?
if someone is shipping a new version of the same function it probably should change topic output or you could have a blue green type of deployment and have both run at the same time with same source and target

Reply:
I've added versioning to functions. If the same function has multiple versions, it currently transforms to each version for now. I'll add a ticket for exploring sending a percentage to a version.